src/main/java/io/strimzi/kafka/bridge/http/HttpBridge.java

```diff
@@ -131,15 +133,17 @@ public void stop(Future<Void> stopFuture) throws Exception {
 
         this.isReady = false;
 
-        //Consumers cleanup
-        this.httpBridgeContext.getHttpSinkEndpoints().forEach((consumerId, httpEndpoint) -> {
-            if (httpEndpoint != null) {
-                httpEndpoint.close();
-            }
-        });
+        // Consumers cleanup
+        Map<String, SinkBridgeEndpoint> consumers = this.httpBridgeContext.getHttpSinkEndpoints();
+
+        for (Map.Entry<String, SinkBridgeEndpoint> consumer: consumers.entrySet()) {
+            if (consumer.getValue() != null)
+                consumer.getValue().close();
+        }
+
         this.httpBridgeContext.getHttpSinkEndpoints().clear();
```

Comment:
Why don't we have a method on the `httpBridgeContext` to closeSinkEndpoints()? Because clearing another object's Map like this seems be a violation of encapsulation.

Reply:
@stanlyDoge ^^